### PR TITLE
Ui: Add option to show/hide console window (Windows-only)

### LINKED ISSUE #1669: Add an option to hide the console window
#### Overview:
Add an option to completely hide the console window.

#### Smaller Details: 
A checkbox within options > settings would be a good place for it.

#### Nature of Request:  
- Addition
  - Add a menu checkbox to completely hide the the console window.

### Why would this feature be useful?
This benefits end users who only want to see the main Ryujinx window, as well as those who are using an aggregating emulator frontend (e.g. Hyperspin, Launchbox) and wish for a cleaner experience. 
It would also reduce the number of support questions regarding harmless console output messages that tend to attract attention.

### PR DESCRIPTION
**Pull request by merryhime**
Add menu item that allows the console window to be displayed or hidden. Windows-only.

~~Is there a nice way to make a menu item only display on Windows?~~ Fixed.

Closes #1669.

### CODE REVIEW

**Comment by EmulationFanatic** [comment]
Supersedes https://github.com/Ryujinx/Ryujinx/pull/2265 ?

**Comment by Shahil-Ayato** [comment]
Hey Mary, I appreciate you completing my PR which was a continuation of a few others haha. I’m glad you chose to take on this PR as I really wanted to complete this but never had the time to so I’ll close mine up. Good luck! 

supersedes #2265.

**Comment by gdkchan** [comment]
Should close https://github.com/Ryujinx/Ryujinx/issues/1669.

File: Ryujinx/Configuration/ConfigurationState.cs [hunk 1]
```diff
@@ -574,6 +581,7 @@ public void LoadDefault()
             Ui.EnableCustomTheme.Value             = false;
             Ui.CustomThemePath.Value               = "";
             Ui.StartFullscreen.Value               = false;
+            Ui.ShowConsole.Value                   = false;
```
Comment (jduncanator) [inline Ryujinx/Configuration/ConfigurationState.cs:584]: I think the default here should be true. We want to retain the existing behaviour.

**Comment by jduncanator** [comment]
My only feedback here is it would be preferred to register an event handler on the `ShowConsole` reactive config object that updates the console display state when changed, especially for something like this that's effectively stateless.

That would ensure any future code that interacts with this setting correctly sets the consoles visibility state, and would allow you to remove the call to `SetConsoleWindowState` in the checkbox toggle handler.

This was how the settings system was supposed to work when it was originally designed.

File: Ryujinx/Ui/Helper/ConsoleHelper.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            if (hWnd == IntPtr.Zero)
+            {
+                Logger.Notice.Print(LogClass.Application, "Attempted to show/hide console window but console window does not exist");
+                return;
+            }
+
+            ShowWindow(hWnd, show ? SW_SHOW : SW_HIDE);
+        }
+
+        [SupportedOSPlatform("windows")]
```
Comment (gdkchan) [inline Ryujinx/Ui/Helper/ConsoleHelper.cs:41]: Might be wrong, but I think you can put it on a single line like `[SupportedOSPlatform("windows"), DllImport("kernel32")]`.

File: Ryujinx/Ui/Helper/ConsoleHelper.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        public static void SetConsoleWindowState(bool show)
+        {
+            if (OperatingSystem.IsWindows())
+            {
+                SetConsoleWindowStateWindows(show);
+            }
+            else if (show == false)
+            {
+                Logger.Notice.Print(LogClass.Application, "OS doesn't support hiding console window");
```
Comment (gdkchan) [inline Ryujinx/Ui/Helper/ConsoleHelper.cs:20]: I think those should probably be warnings. It's not something that should be happening normally.

**Comment by merryhime** [comment]
Addressed comments above I hope.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Review by jduncanator — APPROVED** [review]
Nice work, thanks!

**Review by Thog — APPROVED** [review]

**Review by AcK77 — APPROVED** [review]

**Pull request merged**